cmd/celestia/state.go

```diff
@@ -0,0 +1,449 @@
+		if err != nil {
+			return fmt.Errorf("error parsing a fee:%v", err)
+		}
+
+		gasLimit, err := strconv.ParseUint(args[4], 10, 64)
+		if err != nil {
+			return fmt.Errorf("error parsing a gas limit:%v", err)
+		}
+
+		txResponse, err := client.State.CancelUnbondingDelegation(
+			cmd.Context(),
+			addr.Address.(state.ValAddress),
+			math.NewInt(amount),
+			math.NewInt(height),
+			math.NewInt(fee),
+			gasLimit,
+		)
+		return printOutput(txResponse, err, nil)
+	},
+}
+
+var beginRedelegateCmd = &cobra.Command{
+	Use:   "begin-redelegate [srcAddress, dstAddress, amount, fee, gasLimit]",
+	Short: "Sends a user's delegated tokens to a new validator for redelegation",
+	Args:  cobra.ExactArgs(5),
+	RunE: func(cmd *cobra.Command, args []string) error {
+		client, err := rpcClient(cmd.Context())
+		if err != nil {
+			return err
+		}
+		srcAddr, err := parseAddressFromString(args[0])
+		if err != nil {
+			return fmt.Errorf("error parsing an address:%v", err)
+		}
+
+		dstAddr, err := parseAddressFromString(args[1])
+		if err != nil {
+			return fmt.Errorf("error parsing an address:%v", err)
+		}
+
+		amount, err := strconv.ParseInt(args[2], 10, 64)
+		if err != nil {
+			return fmt.Errorf("error parsing an amount:%v", err)
+		}
+
+		fee, err := strconv.ParseInt(args[3], 10, 64)
+		if err != nil {
+			return fmt.Errorf("error parsing a fee:%v", err)
+		}
+		gasLimit, err := strconv.ParseUint(args[4], 10, 64)
+		if err != nil {
+			return fmt.Errorf("error parsing a gas limit:%v", err)
+		}
+
+		txResponse, err := client.State.BeginRedelegate(
+			cmd.Context(),
+			srcAddr.Address.(state.ValAddress),
+			dstAddr.Address.(state.ValAddress),
+			math.NewInt(amount),
+			math.NewInt(fee),
+			gasLimit,
+		)
+		return printOutput(txResponse, err, nil)
+	},
+}
+
+var undelegateCmd = &cobra.Command{
+	Use:   "undelegate [valAddress, amount, fee, gasLimit]",
+	Short: "Undelegates a user's delegated tokens, unbonding them from the current validator.",
+	Args:  cobra.ExactArgs(4),
+	RunE: func(cmd *cobra.Command, args []string) error {
+		client, err := rpcClient(cmd.Context())
+		if err != nil {
+			return err
+		}
+		addr, err := parseAddressFromString(args[0])
+		if err != nil {
+			return fmt.Errorf("error parsing an address:%v", err)
+		}
+
+		amount, err := strconv.ParseInt(args[1], 10, 64)
+		if err != nil {
+			return fmt.Errorf("error parsing an amount:%v", err)
+		}
+		fee, err := strconv.ParseInt(args[2], 10, 64)
+		if err != nil {
+			return fmt.Errorf("error parsing a fee:%v", err)
+		}
+		gasLimit, err := strconv.ParseUint(args[3], 10, 64)
+		if err != nil {
+			return fmt.Errorf("error parsing a gas limit:%v", err)
+		}
+
+		txResponse, err := client.State.Undelegate(
+			cmd.Context(),
+			addr.Address.(state.ValAddress),
+			math.NewInt(amount),
+			math.NewInt(fee),
+			gasLimit,
+		)
+		return printOutput(txResponse, err, nil)
+	},
+}
+
+var delegateCmd = &cobra.Command{
+	Use:   "delegate [valAddress, amount, fee, gasLimit]",
```

Comment:
```suggestion
	Use:   "delegate [valAddress] [amount] [fee] [gasLimit]",
```

applies for all cmds that take args btw

Reply:
fixed